xdsl/printer.py

```diff
@@ -481,11 +481,24 @@ def print_float(self, value: float, type: AnyFloat):
                 )
         else:
             # to mirror mlir-opt, attempt to print scientific notation iff the value parses losslessly
-            float_str = f"{value:.6e}"
-            if float(float_str) == value:
+            float_str = f"{value:.5e}"
+            index = float_str.find("e")
+            float_str = float_str[:index] + "0" + float_str[index:]
+
```

Comment:
mlir does this interesting thing where it rounds to 5 digits after the comma but still prints the 6th 0